src/data/polynomial/laurent.lean

```diff
@@ -246,6 +247,53 @@ show T n * _ = _, by
 lemma T_mul (n : ℤ) (f : R[T;T⁻¹]) : T n * f = f * T n :=
 (commute_T n f).eq
 
+/--  `trunc : R[T;T⁻¹] →+ R[X]` maps a Laurent polynomial `f` to the polynomial whose terms of
+nonnegative degree coincide with the ones of `f`.  The terms of negative degree of `f` "vanish".
+`trunc` is a left-inverse to `polynomial.to_laurent`. -/
+def trunc : R[T;T⁻¹] →+ R[X] :=
```

Comment:
Can we have `trunc` as an `R`-linear map?

Reply:
I think that most of the pieces are there, but not in a bundled version.  What I mean is that
```lean
def trunc : R[T;T⁻¹] →[R] R[X] :=
begin
  refine (_ : add_monoid_algebra R ℕ →[R] R[X]).comp _,
  { exact ⟨(to_finsupp_iso R).symm, by simp⟩ },
  { refine ⟨λ r, comap_domain _ r (set.inj_on_of_injective (λ a b ab, int.of_nat.inj ab) _), _⟩,
    exact λ r f, comap_domain_smul _ _ _ }
end
```
works, but then there are not nice lemmas to get `simp` to do its magic.  I'd rather leave this for a later PR: adding the `comap_domain.add_monoid_hom` was already painful enough!  :smile: